teachertool/src/utils/index.ts

```diff
@@ -44,3 +45,24 @@ export function getProjectLink(inputText: string): string {
     const hasMakeCode = inputText?.indexOf("makecode") !== -1;
     return hasMakeCode ? inputText : `https://makecode.com/${inputText}`;
 }
+
+export function splitCriteriaTemplate(template: string): CriteriaTemplateSegment[] {
+    // Split by the regex, which will give us an array where every other element is a parameter.
+    const paramRegex = /\$\{([\w\s]+)\}/g;
+    const parts = template.split(paramRegex);
+
+    const segments: CriteriaTemplateSegment[] = [];
+    for (let i = 0; i < parts.length; i++) {
+        const part = parts[i];
+
+        if (part) {
+            segments.push({ type: i % 2 === 0 ? "plain-text" : "param", content: part });
```

Comment:
I believe it is, because `string.split()` will insert an empty string in the returned list if the string starts with the separator. (Documented [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split#:~:text=If%20separator%20appears%20at%20the%20beginning%20(or%20end)%20of%20the%20string%2C%20it%20still%20has%20the%20effect%20of%20splitting%2C%20resulting%20in%20an%20empty%20(i.e.%20zero%20length)%20string%20appearing%20at%20the%20first%20(or%20last)%20position%20of%20the%20returned%20array.)). Included a little more info in this comment: https://github.com/microsoft/pxt/pull/9911#discussion_r1517038306